google-cloud-logging/src/main/java/com/google/cloud/logging/MonitoredResourceUtil.java

```diff
@@ -158,11 +160,9 @@ private static String getValue(Label label, String resourceType) {
         value = MetadataConfig.getClusterName();
         break;
       case ContainerName:
-        if (resourceType.equals("k8s_container")) {
-          String hostName = System.getenv("HOSTNAME");
-          value = hostName.substring(0, hostName.indexOf("-"));
-        } else {
-          value = MetadataConfig.getContainerName();
+        value = MetadataConfig.getContainerName();
+        if (value == null && resourceType.equals("k8s_container")) {
+          value = getContainerNameFromHostName(System.getenv("HOSTNAME"));
```

Comment:
This code retrieves **pod** name and not container name. By default containers receive the same hostname which is a [pod name](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-hostname-and-subdomain-fields) if the pod is launched as standalone or the name is generated following the template `REPLICASET-?????` where `REPLICASET` is a name of the replicaset build as `PODNAME-HASH` and `?????` are 5 alphanumeric identifier.
So, when you extract the first part of the hostname you get the pod name and not container name.